python/external_models/external_parser.py

```diff
@@ -0,0 +1,84 @@
+import os
+import torch
+import argparse
+import numpy as np
+from loguru import logger
+from typing import List, Tuple
+from abc import ABC, abstractmethod
+from transformers import (
+    AutoModelForCausalLM,
+    AutoModelForSeq2SeqLM,
+    AutoTokenizer,
+    AutoModelForTextEncoding,
+)
+
+
+def get_cuda_if_available():
+    return torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+
+def pre_process_input(model_name, input):
+    if model_name == "internlm/internlm2-math-plus-1_8b":
+        prompt="My LEAN 4 state is:\n```lean\n" + input + \
+        "```\nPlease predict a possible tactic to help me prove the theorem."
+        prompt = f"""<|im_start|>user\n{prompt}<|im_end|>\n<|im_start|>assistant\n"""
+    elif model_name == "gpt-3.5-turbo" or model_name == "gpt-4-turbo-preview":
+        prompt = 'Here is a theorom you need to prove in Lean:\n' + \
+            input+'\nNow you should suggest one line tactic in lean code:'
+    elif 'gemini' in model_name  or "claude" in model_name:
+        prompt = 'Here is a theorom you need to prove in Lean:\n' + \
+            input+'\nNow you should suggest one line tactic in lean code:'
+            
+    #implement here        
+    else:
+        assert NotImplementedError
```

Comment:
What does `assert NotImplementedError` do here? Can't we just raise an `NotImplementedError` if the model is something else, or if some explicit error messages are desired here, we can throw a line indicating "external model not supported".

Reply:
Thanks for your advise! I will fix that.